src/libcore/num/mod.rs

```diff
@@ -1401,12 +1401,8 @@ $EndFeature, "
 ```"),
             #[stable(feature = "no_panic_abs", since = "1.13.0")]
             #[inline]
-            pub fn wrapping_abs(self) -> Self {
-                if self.is_negative() {
-                    self.wrapping_neg()
-                } else {
-                    self
-                }
+            pub const fn wrapping_abs(self) -> Self {
+                (self ^ (self >> ($BITS - 1))).wrapping_sub(self >> ($BITS - 1))
```

Comment:
I am not a big fan of making code less readable just to please our const qualification. :/

Cc @oli-obk @eddyb

Reply:
We've had variables for a while, please use them.

Reply:
Has it been verified that this optimizes to identical LLVM IR?

Reply:
@eddyb Since this has already passed final comment period and been merged, shall I open a new PR to use variables?

Reply:
@nikic I can't read LLVM IR; what I did do before submitting the PR was to use llvm-mca, and although the assembly is not identical, the performance measure from llvm-mca was the same, or if I remember well, in some tests the reverse throughput improved by something tiny like 0.1, but not regressed. I only used llvm-mca with i686 or x86-64 instructions, and I don't think I tested all types comprehensively.

Reply:
@tspiteri Asm and IR of the two implementations: https://rust.godbolt.org/z/BXphZX

The new implementation is not combined into a select. The throughput of the final asm is only half the problem. LLVM understands abs operations as a special SelectPatternFlavor and can optimize them accordingly. This will not work if the abs is not in its canonical representation (as a select).

Reply:
@nikic Hmm, using variables as @eddyb suggested seems to hit two birds with one stone. If I insert a variable for the sign, I get one implementation and an alias: https://godbolt.org/z/m-seT0

Does that mean that the performance will not regress?

Reply:
@tspiteri Oops, I simply made a typo in my test and that's why the result is different, duh :) It actually does produce the same IR, both with variable and without. So forget everything I said, this change is fine from the optimization perspective.

Reply:
@nikic While `abs` and `wrapping_abs` seem to be fine performance-wise, `overflowing_abs` from this PR actually does suffer from the problem discussed, so I will *not* forget everything you said! :) It's easily fixed by making it return `(self.wrapping_abs(), self == Self::min_value())`.

Reply:
And also, the code with the sign factored out is more readable. ;)

That said, I still think we should rather wait for CTFE to support conditionals than compromise code readability, and even the `let sign` version is still arcane magic. But that call is up to T-libs.

Reply:
No, we should definitely make as many of these small ops into `const` as soon as we can.